### PR TITLE
0.13: Animatable trait and animation interpolation

### PR DESCRIPTION
**Pull request by alice-i-cecile**
Part of #891. I've merged the two animation features into one PR to make sure that the flow makes sense.

### CODE REVIEW

File: content/news/2024-02-03-bevy-0.13/index.md [hunk 1]
```diff
@@ -1005,17 +1005,61 @@ commands.spawn(NodeBundle {
...
 TODO.
 
-## Animation Interpolation
+## Animation Interpolation Methods
 
-<div class="release-feature-authors">authors: @TODO</div>
+<div class="release-feature-authors">authors: @mockersf</div>
 
-TODO.
+Generally, animations are defined by their **keyframes**: snapshots of the position (and other state) or objects at moments along a time line?
```
Comment (TrialDragon) [inline content/news/2024-02-03-bevy-0.13/index.md:1012]:
```suggestion
Generally, animations are defined by their **keyframes** which are snapshots of the position, other states, or objects at moments along a timeline.
```
Not sure if this change also changes the technical meaning trying to be conveyed. I also presumed the intent was to make a statement, not ask a question.

File: content/news/2024-02-03-bevy-0.13/index.md [hunk 2]
```diff
@@ -1005,17 +1005,61 @@ commands.spawn(NodeBundle {
...
+But what happens between those keyframes? Game engines need to **interpolate** between them: smoothly transitioning from one state to the next.
```
Comment (TrialDragon) [inline content/news/2024-02-03-bevy-0.13/index.md:1013]:
```suggestion
But what happens between those keyframes? Game engines need to **interpolate** between them, smoothly transitioning from one state to the next.
```

File: content/news/2024-02-03-bevy-0.13/index.md [hunk 3]
```diff
@@ -1005,17 +1005,61 @@ commands.spawn(NodeBundle {
...
+
+The simplest interpolation method is linear: just move an equal distance towards the next keyframe every unit of time.
```
Comment (TrialDragon) [inline content/news/2024-02-03-bevy-0.13/index.md:1015]:
```suggestion
The simplest interpolation method is linear where you just move an equal distance towards the next keyframe every unit of time.
```

File: content/news/2024-02-03-bevy-0.13/index.md [hunk 4]
```diff
@@ -1005,17 +1005,61 @@ commands.spawn(NodeBundle {
...
+    /// Implementors should return a default value when no inputs are provided here.
+    fn blend(inputs: impl Iterator<Item = BlendInput<Self>>) -> Self;
+
+    /// Post-processes the value using resources in the [`World`].
+    /// Most animatable types do not need to implement this.
+    fn post_process(&mut self, _world: &World) {}
+}
+```
+
+This is the first step towards animation blending and an asset-driven animation graph: essential for shipping large scale 3D games in Bevy.
```
Comment (TrialDragon) [inline content/news/2024-02-03-bevy-0.13/index.md:1058]:
```suggestion
This is the first step towards animation blending and an asset-driven animation graph; essential for shipping large scale 3D games in Bevy.
```

File: content/news/2024-02-03-bevy-0.13/index.md [hunk 5]
```diff
@@ -1005,17 +1005,61 @@ commands.spawn(NodeBundle {
...
+But for now, this is a building block: we've implemented this for a few key types (`Transform`, `f32` and `glam`'s `Vec` types) and published the trait.
```
Comment (TrialDragon) [inline content/news/2024-02-03-bevy-0.13/index.md:1059]:
```suggestion
But for now, this is just a building block. We've implemented this for a few key types (`Transform`, `f32` and `glam`'s `Vec` types) and published the trait.
```

**Review by TrialDragon — CHANGES_REQUESTED** [review]
Minor grammar changes, otherwise LGTM.

File: content/news/2024-02-03-bevy-0.13/index.md (hunk 1 above)
Comment (alice-i-cecile) [reply content/news/2024-02-03-bevy-0.13/index.md:1012]:
```suggestion
Generally, animations are defined by their **keyframes**: snapshots of the position (and other state) or objects at moments along a time line.
```

File: content/news/2024-02-03-bevy-0.13/index.md (hunk 3 above)
Comment (alice-i-cecile) [reply content/news/2024-02-03-bevy-0.13/index.md:1015]:
```suggestion
The simplest interpolation method is linear: the object just moves an equal distance towards the next keyframe every unit of time.
```

**Pull request merged**